### DISCUSSION TITLE
does Longhorn  support striping？

### DISCUSSION THREAD

**Discussion by hchyue** [longhorn/longhorn#9373] [General]
Does Longhorn natively support striping, which is a technique where data is split across multiple disks to improve performance?

**Comment by james-munson** [comment]
Longhorn does not natively use RAID 0, aka striping.  It does do mirroring (RAID 1) when the user configures volumes with more than 1 replica, but that is more for redundancy than for performance (although it can provide a benefit in read performance as well.)

Striping could be done at a level below Longhorn, if the block device configured for Longhorn to use were a striped collection of drives assembled in a logical volume managed by LVM, say.  The performance gain over a single drive would be hard to detect when layered with iSCSI and network access from nodes within a cluster.

**Comment by hchyue** [comment]
does Longhorn plan to support striping natively？

**Comment by james-munson** [comment]
I think so as a long-term goal - see `[FEATURE] v2 Data Engine Sharding:`  https://github.com/longhorn/longhorn/issues/1061.

**Comment by hchyue** [comment]
i see
